### PR TITLE
Update: fix services (on CentOS)

### PR DESCRIPTION
**Pull request by till**
- we can't configure docker with daemon.json
 - so we add overrides

### CODE REVIEW

**Comment by atosatto** [comment]
Hi @till thanks for this PR.

Which version of CentOS are you running on?
Can you please do a `systemctl cat docker` and add some more information about the issue you found with configuring docker with `daemon.json`?

**Comment by till** [comment]
@atosatto 

Tried to install Docker `18.09.2` on CentOS7 — the services are this:

```
# /usr/lib/systemd/system/docker.service
[Unit]
Description=Docker Application Container Engine
Documentation=https://docs.docker.com
BindsTo=containerd.service
After=network-online.target firewalld.service
Wants=network-online.target
Requires=docker.socket

[Service]
Type=notify
# the default is not to use systemd for cgroups because the delegate issues still
# exists and systemd currently does not support the cgroup feature set required
# for containers run by docker
ExecStart=/usr/bin/dockerd -H fd://
ExecReload=/bin/kill -s HUP $MAINPID
TimeoutSec=0
RestartSec=2
Restart=always

# Note that StartLimit* options were moved from "Service" to "Unit" in systemd 229.
# Both the old, and new location are accepted by systemd 229 and up, so using the old location
# to make them work for either version of systemd.
StartLimitBurst=3

# Note that StartLimitInterval was renamed to StartLimitIntervalSec in systemd 230.
# Both the old, and new name are accepted by systemd 230 and up, so using the old name to make
# this option work for either version of systemd.
StartLimitInterval=60s

# Having non-zero Limit*s causes performance problems due to accounting overhead
# in the kernel. We recommend using cgroups to do container-local accounting.
LimitNOFILE=infinity
LimitNPROC=infinity
LimitCORE=infinity

# Comment TasksMax if your systemd version does not supports it.
# Only systemd 226 and above support this option.
TasksMax=infinity

# set delegate yes so that systemd does not reset the cgroups of docker containers
Delegate=yes

# kill only the docker process, not all processes in the cgroup
KillMode=process

[Install]
WantedBy=multi-user.target

# /etc/systemd/system/docker.service.d/override.conf
# Ansible managed: Do NOT edit this file manually!
 [Service]
 ExecStart=
 ExecStart=/usr/bin/dockerd
```

My override ensures `ExecStart` is only `/usr/bin/dockerd`. Since their unit provides `-H fd://` it doesn't allow me to set/add hosts via `/etc/docker/daemon.json`. I backported this fix from our `geerlingguy.docker` fork, since we found your role to be a little more straight forward and closer to what we need (a Swarm setup).

I am doing this, because we sometimes need to expose Docker via it's http/https API. It's not great, but when we do, we add it via `hosts` in `/etc/docker/daemon.json`.

**Comment by till** [comment]
@atosatto Ping?

File: tasks/setup-swarm-cluster.yml [hunk 1]
```diff
@@ -84,4 +84,42 @@
     and 'docker_swarm_manager' in group_names
     and inventory_hostname != groups['docker_swarm_manager'][0]"
   tags:
+<<<<<<< HEAD:tasks/setup-swarm-cluster.yml
```
Comment (till) [inline tasks/setup-swarm-cluster.yml]: @rwos :D

Comment (rwos) [reply tasks/setup-swarm-cluster.yml]: sorry, fixed

**Comment by atosatto** [comment]
@till I've implemented an alternative solution to this issue which would allow to optionally override the systemd unit for both containerd and Docker in #64. Would you mind having a look to that PR and see whether the solution might work for your use-case?

**Comment by till** [comment]
Could have waited a little, since this PR was sitting here for 3 months? ;)

**Comment by atosatto** [comment]
I had to release v2.0.0. We can do a v2.0.1 for any change or feedback.

**Pull request closed without merging**